dags/check_brokenreaders.py

```diff
@@ -0,0 +1,81 @@
+from airflow import DAG
+from datetime import datetime, timedelta
+from airflow.operators.python_operator import PythonOperator
+from airflow.hooks.postgres_hook import PostgresHook
+from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+from psycopg2 import sql
+from psycopg2.extras import execute_values
+from psycopg2 import connect, Error
+import logging
+import configparser
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+#To connect to pgadmin
+CONFIG = configparser.ConfigParser()
+CONFIG.read(r'/home/mohan/cre.cfg')
+dbset = CONFIG['DBSETTINGS']
+con = connect(**dbset)
+
+def bad_readers(con, run_date):
+    with con.cursor() as cursor: 
+        select_query = """SELECT reader from mohan.broken_reader()"""
+        cursor.execute(select_query)
+        while True:
+            broken_reader = cursor.fetchall()
+            num_readers = len(broken_reader)
+            if num_readers > 0:
+                LOGGER.info(broken_reader)
+                raise Exception (num_readers, ' readers are broken as of', run_date)
+            if num_readers == 0:
+                LOGGER.info('All readers are working as of date = %s', run_date')
+                
+SLACK_CONN_ID = 'slack'
+def task_fail_slack_alert(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+    slack_msg = """
+            :red_circle: A bt Reader may be broken!
+            *Task*: {task}  
+            *Dag*: {dag} 
+            *Execution Time*: {exec_date}  
+            *Log Url*: {log_url} 
```

Comment:
This should be updated to our more recent Slack message style.

See for example

https://github.com/CityofToronto/bdit_data-sources/blob/master/dags/pull_wys.py#L28-L44